File: frontend/src/app/services/child/models/Child.ts
export interface Child {
  id: string;
  name: string;
  age: string;
  clothesSize: string;
  shoeSize: string;
  legalResponsible: string;
  familyAcronym: string;
  familyPhone: string;
  familyAddress: string;
  godParents: GodParent[];
}

export interface GodParent {
  id: string | undefined;
  name: string;
  phone: string;
  isClothesSelected: boolean;
  isShoeSelected: boolean;
  isGiftSelected: boolean;
}

export interface DashChildModel {
  id: string;
  name: string;
  familyCode: string;
  deliveredGifts: number;
  remainingGifts: number;
}

export interface TypeResponse {
  id: number;
  description: string;
}

export interface GodParentResponse {
  id: string;
  name: string;
  contactNumber: string;
  address: string;
}

export interface FamilyChildResponse {
  id: string;
  code: string;
  contactNumber: string;
  address: string;
  member: string;
}

export interface GiftResponse {
  isDelivered: boolean;
  giftType: TypeResponse;
  godParent: GodParentResponse;
}

export interface GetChildrenByIdResponse {
  id: string;
  name: string;
  age: string;
  clotheSize: string;
  shoeSize: string;
  family: FamilyChildResponse;
  genre: TypeResponse;
  gifts: GiftResponse[];
}

export interface CreateGiftRequest {
  childId: string;
  typeId: number;
  godParent: Partial<GodParentResponse>;
}

export interface CreateChildRequest {
  id: string;
  familyId: string;
  name: string;
  age: string;
  clotheSize: string;
  shoeSize: string;
  genre: number;
}

File: backend/Jcc.Api/JccApi/Entities/Child.cs
using System;
using System.Collections.Generic;

namespace JccApi.Entities
{
    public class Child
    {
        protected Child() { }

        public Child(Guid id)
        {
            Id = id;
        }

         public Child(string name, string age, string clotheSize, string shoeSize, int genreTypeId, Guid familyId)
        {
            Id = Guid.NewGuid();
            Name = name;
            Age = age;
            ClotheSize = clotheSize;
            ShoeSize = shoeSize;
            GenreTypeId = genreTypeId;
            FamilyId = familyId;
        }

        public Child(Guid id, string name, string age, string clotheSize, string shoeSize, int genreTypeId, Guid familyId)
        {
            Id = id;
            Name = name;
            Age = age;
            ClotheSize = clotheSize;
            ShoeSize = shoeSize;
            GenreTypeId = genreTypeId;
            FamilyId = familyId;
        }

        public Guid Id { get; private set; }
        public string Name { get; private set; }
        public string Age { get; private set; }
        public string ClotheSize { get; private set; }
        public string ShoeSize { get; private set; }
        public int GenreTypeId { get; private set; }
        public Guid FamilyId { get; private set; }

        public Genre GenreType { get; private set; }
        public Family Family { get; private set; }
        public ICollection<Gift> Gifts { get; private set; }
    }
}
